sympy/combinatorics/perm_groups.py

```diff
@@ -694,6 +694,63 @@ def basic_transversals(self):
             self.schreier_sims()
         return self._transversals
 
+    def composition_series(self):
+        """
+        Return the composition series for a group as a list
+        of permutation groups.
+
+        A composition series is a subnormal series such that each
+        factor group `H(i+1) / H(i)` is simple.
+        A subnormal series is a composition series only if it is of
+        maximum length.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.named_groups import (AlternatingGroup,
+        ... SymmetricGroup)
+        >>> S = SymmetricGroup(4)
+        >>> series = S.composition_series()
+        >>> len(series)
+        5
+        >>> A = AlternatingGroup(4)
+        >>> series[1] == A
+        True
+        >>> series[2].is_subgroup(A)
+        True
+        >>> len(A.composition_series())
+        4
+
+        """
+        if not self.is_solvable:
+            raise NotImplementedError('Group should be solvable')
+        der = self.derived_series()
+        series = [self]
+
+        for i in range(len(der)-1):
+            comp = []
+            low = der[i+1]
+            gns = low.generators
+            for elm in der[i].generators:
+                if low == der[i]:
```

Comment:
Is it guaranteed that `low` and `der[i]` will always be `number` and not expressions containing `Symbol`?

Reply:
They are neither numbers nor expressions, they are perm groups.